src/RouterInterface.php

```diff
@@ -28,11 +29,10 @@
      * or `generateUri()` have already been called, to ensure integrity of the
      * router between invocations of either of those methods.
      *
-     * @param Route $route
      * @throws Exception\RuntimeException when called after match() or
      *     generateUri() have been called.
      */
-    public function addRoute(Route $route);
+    public function addRoute(Route $route) : void;
```

Comment:
Ooof. The signature changes in this interface will require new major versions of each router *unless we update them to have return values now* (return types are covariant, so an implementation can define a more specific return type).

I'll experiment with that momentarily on the zendrouter repo.